torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3049,6 +3137,34 @@ def _finalize_params(fsdp_module: FullyShardedDataParallel) -> None:
                     # reset this flag for cases like "one forward pass + multiple backward passes"
                     self._post_backward_callback_queued = False
 
+        if (self.use_param_exec_order_policy() and self._param_exec_order_prep_stage):
+            self._param_exec_order_prep_stage = False
```

Comment:
nit: also put these logic into helper function, as we are branching the logic under the if condition